src/Evolution/Systems/ScalarWave/Characteristics.hpp

```diff
@@ -0,0 +1,226 @@
+#include "Domain/FaceNormal.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <typename>
+class Variables;
+
+namespace Tags {
+template <typename Tag>
+struct Normalized;
+}  // namespace Tags
+/// \endcond
+
+namespace ScalarWave {
+namespace Tags {
+struct ConstraintGamma2Compute : ConstraintGamma2, db::ComputeTag {
+  using argument_tags = tmpl::list<Psi>;
+  static auto function(const Scalar<DataVector>& psi) noexcept {
+    return make_with_value<type>(psi, 0.);
+  }
+  using base = ConstraintGamma2;
+};
+}  // namespace Tags
+// @{
+/*!
+ * \ingroup ScalarWave
+ * \brief Compute the characteristic speeds for the scalar wave system.
+ *
+ * Computes the speeds as described in "Optimal constraint projection for
+ * hyperbolic evolution systems" by Holst et. al \cite Holst2004wt
+ * [see text following Eq.(32)]. The characteristic fields' names used here
+ * are similar to the paper:
+ *
+ * \f{align*}
+ * \mathrm{SpECTRE} && \mathrm{Holst} \\
+ * u^{\psi} && Z^1 \\
+ * u^0_{i} && Z^{2}_{i} \\
+ * u^{\pm} && u^{1\pm}
+ * \f}
+ *
+ * The corresponding characteristic speeds \f$\lambda_\alpha\f$ are given in
+ * the text following Eq.(38) of \cite Holst2004wt :
+ *
+ * \f{align*}
+ * \lambda_{\psi} =& 0 \\
+ * \lambda_{0} =& 0 \\
+ * \lambda_{\pm} =& \pm 1
+ * \f}
+ *
+ * where \f$n_k\f$ is the unit normal to the surface.
+ */
+template <size_t Dim>
+std::array<DataVector, 4> characteristic_speeds(
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        unit_normal_one_form) noexcept;
+
+template <size_t Dim>
+void characteristic_speeds(
+    gsl::not_null<std::array<DataVector, 4>*> char_speeds,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        unit_normal_one_form) noexcept;
+
+template <size_t Dim>
+struct CharacteristicSpeedsCompute : Tags::CharacteristicSpeeds<Dim>,
+                                     db::ComputeTag {
+  using base = Tags::CharacteristicSpeeds<Dim>;
+  using type = typename base::type;
+  using argument_tags =
+      tmpl::list<::Tags::Normalized<::Tags::UnnormalizedFaceNormal<Dim>>>;
+
+  static typename Tags::CharacteristicSpeeds<Dim>::type function(
+      const tnsr::i<DataVector, Dim, Frame::Inertial>&
+          unit_normal_one_form) noexcept {
+    return characteristic_speeds(unit_normal_one_form);
+  };
+};
+// @}
+
+// @{
+/*!
+ * \ingroup ScalarWave
+ * \brief Computes characteristic fields from evolved fields
+ *
+ * \ref CharacteristicFieldsCompute and
+ * \ref EvolvedFieldsFromCharacteristicFieldsCompute convert between
+ * characteristic and evolved fields for the scalar-wave system.
+ *
+ * \ref CharacteristicFieldsCompute computes
+ * characteristic fields as described in "Optimal constraint projection for
+ * hyperbolic evolution systems" by Holst et. al \cite Holst2004wt .
+ * Their names used here differ from this paper:
```

Comment:
Wait, the comment above (for the speeds) says the names are similar. But the tables are the same... am I missing something?

Reply:
They are *different*, it was a typo above which is now fixed